### PR TITLE
Issue-301: adding more distance metric types for kneighbors

### LINKED ISSUE #301: Add support for more distance metric types in Sklearn KNeighbors classifier
Add support for the following distance metric types:

identifier | class name | args | distance function
-- | -- | -- | --
“manhattan” | ManhattanDistance |   | sum(\|x - y\|)
“chebyshev” | ChebyshevDistance |   | max(\|x - y\|)
“wminkowski” | WMinkowskiDistance | p, w | sum(\|w * (x - y)\|^p)^(1/p)
“seuclidean” | SEuclideanDistance | V | sqrt(sum((x - y)^2 / V))
“mahalanobis” | MahalanobisDistance | V or VI | sqrt((x - y)' V^-1 (x - y))

### PR DESCRIPTION
**Pull request by scnakandala**
Closes #301

### CODE REVIEW

**Comment by interesaaat** [comment]
Is this good for review?

**Comment by scnakandala** [comment]
> Is this good for review?

yes. There is an issue with the way how pytorch handles infinity norm. So I commented out some tests. I will create a different issue for that.

**Comment by interesaaat** [comment]
Thanks Supun!
